app/server/skychat/commands/impl/core/MessageHistoryPlugin.ts

```diff
@@ -42,10 +44,9 @@ export class MessageHistoryPlugin extends Plugin {
 
             // Randomly add a sticker
             if (Math.random() < .7) {
-                const formatter = MessageFormatter.getInstance();
-                const stickers = Object.keys(formatter.stickers);
-                const sticker = stickers[Math.floor(stickers.length * Math.random())];
-                fakeText += ' ' + sticker;
+                const stickers = Object.keys(this.formatter.stickers);
+                const sticker = stickers.length ? ' ' + stickers[Math.floor(stickers.length * Math.random())] : '';
+                fakeText += sticker;
             }
```

Comment:
Thanks for the fix! How about pushing the `stickers.length` check at the higher level to avoid the ternary operator, like this:
```typescript
// Randomly add a sticker if stickers are defined
if (stickers.length && Math.random() < .7) {
    const stickers = Object.keys(this.formatter.stickers);
    fakeText += ' ' + stickers[Math.floor(stickers.length * Math.random())];
}
```

Seems even cleaner, would you mind changing it?

Reply:
Genius, seems even cleaner, but how do you check for `stickers` length in the condition when you define it in the instruction just after ?
    ```
const stickers = Object.keys(this.formatter.stickers);
```

Reply:
Right 🙃, read it too fast, it can be put just before the `for` on line 39-40. This will avoid re-computing the list for every faked message.

Reply:
lol yes, reading too fast is typical student error.
I've done the fix 👍